--- /home/shoops/code/ENISI/deps/src/project_repast/INSTALLATION/install.sh	2013-08-13 11:15:31.000000000 -0400
+++ install.sh	2016-11-11 15:25:16.470829000 -0500
@@ -7,8 +7,8 @@
 BASE_DIR=$PWD/ext/
 cd INSTALLATION
 
-MPI_COMPILER_INVOCATION=$BASE_DIR/MPICH/bin/mpicxx
-
+MPI_CXX_COMPILER="${2:-${BASE_DIR}/MPICH/bin/mpicxx}"
+MPI_CXX_LINK_FLAGS="${3:--static}"
 
 # cURL
 if [[ $1 == *lcurl* ]]
@@ -115,7 +115,7 @@
   cp -r ./boost_1_54_0/boost $BASE_DIR/Boost/Boost_1.54/include
   cd boost_1_54_0
   ./bootstrap.sh --prefix=$BASE_DIR/Boost/Boost_1.54/ --with-libraries=system,filesystem,mpi,serialization
-  echo "using mpi : $MPI_COMPILER_INVOCATION ;" >>./tools/build/v2/user-config.jam
+  echo "using mpi : $MPI_CXX_COMPILER ;" >>./tools/build/v2/user-config.jam
   ./b2 --layout=tagged link=static variant=release threading=multi runtime-link=static stage install
   cd ..
 fi
@@ -123,6 +123,6 @@
 # Repast HPC
 if [[ $1 == *rhpc* ]]
 then
-  make -f Makefile CXX=$MPI_COMPILER_INVOCATION CXXLD="$MPI_COMPILER_INVOCATION -static" BOOST_INCLUDE_DIR=$BASE_DIR/Boost/Boost_1.54/include BOOST_LIB_DIR=$BASE_DIR/Boost/Boost_1.54/lib BOOST_INFIX=-mt-s NETCDF_INCLUDE_DIR=$BASE_DIR/NetCDF/include NETCDF_LIB_DIR=$BASE_DIR/NetCDF/lib CURL_INCLUDE_DIR=$BASE_DIR/CURL/include CURL_LIB_DIR=$BASE_DIR/CURL/lib
+  make -f Makefile CXX="${MPI_CXX_COMPILER}" CXXLD="$MPI_CXX_COMPILER $MPI_CXX_LINK_FLAGS" BOOST_INCLUDE_DIR=$BASE_DIR/Boost/Boost_1.54/include BOOST_LIB_DIR=$BASE_DIR/Boost/Boost_1.54/lib BOOST_INFIX=-mt-s NETCDF_INCLUDE_DIR=$BASE_DIR/NetCDF/include NETCDF_LIB_DIR=$BASE_DIR/NetCDF/lib CURL_INCLUDE_DIR=$BASE_DIR/CURL/include CURL_LIB_DIR=$BASE_DIR/CURL/lib
 fi
 
